### PR TITLE
Use last windowed input ids to update logits

### PR DESCRIPTION
**Pull request by baijumeswani**
Bug: When using windowing logic for large sequence length inputs (sequence length > window size), `Logits::Update` was using the `input_ids` associated with the first chunk (always) since it didn't receive any of the subsequent chunks to process.

As a result, for large sequence length inputs (sequence length > window size), incorrect logits were being used to compute the next tokens (always window_size -1). This pull-request addresses this bug by processing the correct windowed input ids.

### CODE REVIEW

File: src/models/decoder_only_pipeline.cpp [hunk 1]
```diff
@@ -357,6 +357,8 @@ DeviceSpan<float> DecoderOnlyPipelineState::Run(int total_length, DeviceSpan<int
       input_ids_->Update(next_tokens);
       UpdateKeyValueCache(next_indices, total_length);
       position_inputs_->Update(next_tokens, total_length, static_cast<int>(input_ids_->GetShape()[1]));
+      logits_.Update(WrapTensor<int32_t>(*model_.p_device_inputs_, *input_ids_->Get()),
```
Comment (apsonawane) [inline src/models/decoder_only_pipeline.cpp:360]: Why do we need to update logits here again after UpdateInputsOutputs?

**Review by aciddelgado — APPROVED** [review]

File: src/models/decoder_only_pipeline.cpp (hunk 1 above)
Comment (baijumeswani) [reply src/models/decoder_only_pipeline.cpp:360]: This is for windowing logic. For windowing, the prompt is broken down into window chunks. Each window runs through the model. The very last chunk can have input_ids that are padded. In such a case, we want to use the logits of the last valid token in the input ids and not the last one from the window. That's why we need the update here.
